### ISSUE TITLE
Segfault in backward for model with grouped convolution (on CPU)

### ISSUE THREAD

**Issue by pbloem** [pytorch/pytorch#25994]
## 🐛 Bug

When training a particular model (see below) on CPU, a segfault is thrown when ```loss.backward()``` is called. 

## To Reproduce
Here's the smallest example I could manage:

```python 
import torch
from torch import nn
from torch.autograd import Variable
import math

class Flatten(nn.Module):

    def forward(self, input):
        return input.view(input.size(0), -1)

class ResBlock(nn.Module):
    """
    Inverted residual block (as in mobilenetv2)
    """
    def __init__(self,  c, wide=None, kernel=3, grouped=True):
        super().__init__()

        wide = 6 * c if wide is None else wide
        padding = int(math.floor(kernel/2))

        self.convs = nn.Sequential(
            nn.Conv2d(c, wide, kernel_size=1),
            nn.Conv2d(wide, wide, kernel_size=kernel, padding=padding, groups=wide if grouped else 1),
            nn.Conv2d(wide, c, kernel_size=1),
            nn.BatchNorm2d(c), nn.ReLU()
        )

    def forward(self, x):

        return self.convs(x) + x # wo the skip, the segfault happens immediately


model = nn.Sequential(
            nn.Conv2d(3, 32, kernel_size=3, padding=1),
            nn.MaxPool2d(kernel_size=2),
            ResBlock(32, grouped=True),  nn.Conv2d(32, 16, kernel_size=1),
            ResBlock(16, grouped=False), nn.Conv2d(16, 16, kernel_size=1),
            nn.MaxPool2d(kernel_size=16),
            Flatten(),
            nn.Linear(16, 10),
            nn.Softmax(dim=-1)
        )

opt = torch.optim.SGD(lr=0.000001, params=model.parameters()) # SGD and Adam both segfault

torch.manual_seed(0)

for i in range(1000):

    print(i)

    opt.zero_grad()

    x = Variable(torch.randn(64, 3, 32, 32))
    x = model(x)
    loss = x.sum()

    loss.backward() # segfault here

    opt.step()

```
## Expected behavior
No segfault.

## Environment
```
Collecting environment information...
PyTorch version: 1.2.0
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.14.6
GCC version: Could not collect
CMake version: version 3.11.1

Python version: 3.6
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip3] numpy==1.15.1
[pip3] torch==1.2.0
[pip3] torchtext==0.4.0
[pip3] torchvision==0.2.1
[conda] torch                     1.2.0                     <pip>

```

## Additional context

The model "trains" for a few batches before the segfault (two batches in the example above). A certain amount of model complexity seems to be required to reproduce; no layer or block causes the problem by itself. Gradients and output seem stable before the problem occurs.

I think the problem may be in the grouped convolution, but it's difficult to isolate the problem.

Removing skip connections exacerbates the problem.

No problems on GPU in a linux environment.

cc @ezyang @gchanan @zou3519 @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh

**Comment by vpirogov** [comment]
@pbloem,

Thank you for the report. There are a few known issues in Pytorch 1.2 addressed in #25757. Could you please collect the log with `MKLDNN_VERBOSE=1` to capture convolution shapes?

**Comment by XiaobingSuper** [comment]
@pbloem , I can't reproduce your problem in my side, I find your environment have three different torch: 0.4.1, 1.2.0 and 0.4.0, that is very confused. Can you create a new environment by conda and reinstall PyTorch? Please tell me if you still have this problem, thanks!

**Comment by XiaobingSuper** [comment]
@pbloem, thank for your report, I can reproduce it now on macos for the machine with only supoort avx2, but it can be passed on maxos for the machine which cpu supports avx512. For linux environment, the test case can be passed for avx2 machine and avx512 machine, let me check the details reasons, and I will tell you when we solve it. Thanks!

**Comment by XiaobingSuper** [comment]
@pbloem , I test the test case using pytorch-nightly, it can be passed. Can you help verify it again? For your macos environment, you can install pytorch nightly version by:
`conda install pytorch torchvision -c pytorch-nightly` or `pip install numpy && pip install --pre torch torchvision -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html`

**Comment by zou3519** [comment]
High pri to repro this on master.

**Comment by pbloem** [comment]
@XiaobingSuper Just checked nightly and the problem is gone there. Thanks!

**Comment by soumith** [comment]
thanks for letting us know @pbloem and thanks for the bug report
